tests/test_utils.py

```diff
@@ -899,10 +903,224 @@ def test_companion_mass_error_noquantity_Pb():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(u.UnitsError):
+        omdot(Mp, Mc, Pb, e * u.s)
+
+
+def test_omdot_error_noquantity_mp():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        omdot(Mp.value, Mc, Pb, e)
+
+
+def test_omdot_error_noquantity_mc():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        omdot(Mp, Mc.value, Pb, e)
+
+
+def test_omdot_error_noquantity_pb():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        omdot(Mp, Mc, Pb.value, e)
+
+
+def test_pbdot_error_noquantity_mp():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        pbdot(Mp.value, Mc, Pb, e)
+
+
+def test_pbdot_error_noquantity_mc():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        pbdot(Mp, Mc.value, Pb, e)
+
+
+def test_pbdot_error_noquantity_pb():
+    Mp = 1.3381 * u.Msun
+    Mc = 1.2489 * u.Msun
+    Pb = 0.10225156248 * u.d
+    e = 0.0877775
+    with pytest.raises(TypeError):
+        pbdot(Mp, Mc, Pb.value, e)
+
+
+@given(
+    floats(min_value=1.0, max_value=3.0),
+    floats(min_value=0.01, max_value=10),
+    floats(min_value=0.04, max_value=1000),
+    floats(min_value=0.1, max_value=90),
+)
+def test_a1sini_Mc(Mp, Mc, Pb, i):
+    """test a1sini by looking for consistency with companion mass calculation."""
+    Mp = Mp * u.Msun
+    Mc = Mc * u.Msun
+    Pb = Pb * u.d
+    i = i * u.deg
+    x = a1sini(Mp, Mc, Pb, i=i)
+    assert np.isclose(Mc, companion_mass(Pb, x, i=i, mp=Mp))
+
+
+@given(
+    floats(min_value=1.0, max_value=3.0),
+    floats(min_value=0.01, max_value=10),
+    floats(min_value=0.04, max_value=1000),
+    floats(min_value=0.1, max_value=90),
+)
+def test_a1sini_Mp(Mp, Mc, Pb, i):
+    """test a1sini by looking for consistency with pulsar mass calculation."""
+    Mp = Mp * u.Msun
+    Mc = Mc * u.Msun
+    Pb = Pb * u.d
+    i = i * u.deg
+    x = a1sini(Mp, Mc, Pb, i=i)
+    assert np.isclose(Mp, pulsar_mass(Pb, x, Mc, i))
+
+
+@given(
+    arrays(float, 10, elements=floats(0.5, 3)),
+    arrays(float, 10, elements=floats(0.01, 10)),
+    arrays(float, 10, elements=floats(0.04, 1000)),
+    arrays(float, 10, elements=floats(0.1, 90)),
+)
+def test_a1sini_Mc_array(Mp, Mc, Pb, i):
+    """test a1sini by looking for consistency with companion mass calculation with array input."""
+    Mp = Mp * u.Msun
+    Mc = Mc * u.Msun
+    Pb = Pb * u.d
+    i = i * u.deg
+    x = a1sini(Mp, Mc, Pb, i=i)
+    assert (np.isclose(Mc, companion_mass(Pb, x, i=i, mp=Mp))).all()
```

Comment:
`numpy.testing.allclose(thing, other_thing)` provides better error reporting if the things aren't the same shape or aren't all close.

Reply:
good to know.